dbt/adapters/databricks/impl.py

```diff
@@ -86,7 +94,9 @@ def list_relations_without_caching(
     ) -> List[DatabricksRelation]:
         kwargs = {"schema_relation": schema_relation}
         try:
-            results = self.execute_macro(LIST_RELATIONS_MACRO_NAME, kwargs=kwargs)
+            # The catalog for `show table extended` needs to be the current catalog.
```

Comment:
We cannot pass the catalog name in the `SHOW TABLE EXTENDED` command? Something like `SHOW TABLE EXTENDED IN catalog.db LIKE '*'` where `catalog` is not the current catalog.

Reply:
Unfortunately, it's not supported. We need to set the current catalog the same as the catalog in the statement.

```
>>> with dbsql.connect(host, path, token, catalog="main") as conn:
...   with conn.cursor() as cursor:
...     cursor.execute("SHOW TABLE EXTENDED IN alternative.default LIKE '*'")
...     print(cursor.fetchall())
...
Traceback (most recent call last):
...
databricks.sql.exc.ServerOperationError: Error running query: org.apache.spark.sql.AnalysisException: Fail to execute the command as the target schema `alternative.default` is not in the current catalog. Please set the current catalog with 'USE CATALOG alternative' first.
```

Reply:
```suggestion
            # The catalog for `show table extended` needs to match the current catalog.
```